### PR TITLE
more itertools defined

### PR DESCRIPTION
**Pull request by AlexEKoren**
Added cycle, dropwhile, from_iterable, ifilter, ifilterfalse, and takewhile in itertools

### CODE REVIEW

**Comment by AlexEKoren** [comment]
I signed it!

**Comment by ns-cweber** [comment]
Might be good to have some test cases as well.

**Comment by AlexEKoren** [comment]
Done! Thanks for the feedback

**Comment by ns-cweber** [comment]
Looks like you've committed some `.pyc` files by accident?

File: lib/itertools_test.py [hunk 1]
```diff
@@ -16,6 +16,70 @@
...
+  limit = 10 * len(arg)
+  counter = 0
+  for x in itertools.cycle((0, 1, 2)):
+    got.append(x)
+    counter += 1
+    if counter == limit:
+      break
+  assert tuple(got) == want, 'tuple(cycle%s) == %s, want %s' % (arg, tuple(got), want)
+
+def testDropwhile():
```
Comment (ns-cweber) [inline lib/itertools_test.py]: `def TestDropwhile`?

**Comment by AlexEKoren** [comment]
Good catch. Should be fixed. Changed the .gitignore as well.

**Comment by ns-cweber** [comment]
Good work. This all looks good to me, save for the aforementioned typo.

**Comment by AlexEKoren** [comment]
Awesome! I signed the CLA, but not sure why it's not going through. What can I do rectify?

**Comment by trotterdylan** [comment]
Really appreciate you taking this on!

Not sure what's up with the CLA issue, but did you see this comment? https://github.com/google/grumpy/pull/15#issuecomment-270507818

Do you have multiple emails associated with your GitHub account? If so, then make sure that the email that's associated with the git commits is the same one on the CLA?

**Comment by AlexEKoren** [comment]
Yep, I double checked. Same e-mail (capitalization is different, but Github won't let me change it. Guessing that's not the issue)

**Comment by trotterdylan** [comment]
Strangely, it looks like one of your commits is not like the other: 

9sarmkau43u

The one by AlexEKoren is fine, but the others aren't?

**Comment by AlexEKoren** [comment]
@trotterdylan my CLI was setup with another e-mail account. Rewrote old commits with correct credentials! Wow, that was a weird one, great catch.

**Comment by trotterdylan** [comment]
Awesome. This all looks great. Can you make sure that "make precommit" is clean?

**Comment by AlexEKoren** [comment]
Just to make sure I understand correctly: Run 'make precommit' in the root directory and make sure it runs without error?

**Comment by AlexEKoren** [comment]
I get the following output. Thoughts? Tried to make clean as well.

build/src/grumpy/builtin_types.go:587: z.Text undefined (type big.Int has no field or method Text)
build/src/grumpy/long.go:131: x.Text undefined (type *big.Int has no field or method Text)
build/src/grumpy/long.go:281: toLongUnsafe(o).value.Text undefined (type big.Int has no field or method Text)
build/src/grumpy/long.go:289: toLongUnsafe(o).value.Text undefined (type big.Int has no field or method Text)
make: *** [build/pkg/darwin_amd64/grumpy.a] Error 1

**Comment by trotterdylan** [comment]
That's not a problem with your code. It may be that you have an older Go toolchain. Can you paste the output of "go version"? Mine is:

$ go version
go version go1.7.4 darwin/amd64

**Comment by AlexEKoren** [comment]
Oops! Was a while since I updated Go. 

'make precommit' runs with exit status 0!

**Comment by trotterdylan** [comment]
Awesome! Thanks for working on this!

**Pull request merged**